src/runtime/local/kernels/Map.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 The DAPHNE Consortium
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SRC_RUNTIME_LOCAL_KERNELS_MAPOP_H
+#define SRC_RUNTIME_LOCAL_KERNELS_MAPOP_H
+
+#include <runtime/local/context/DaphneContext.h>
+#include <runtime/local/datastructures/DataObjectFactory.h>
+#include <runtime/local/datastructures/DenseMatrix.h>
+
+#include <algorithm>
+
+#include <cassert>
+
+// ****************************************************************************
+// Struct for partial template specialization
+// ****************************************************************************
+
+template<class DTRes, class DTArg>
+struct Map {
+    static void apply(DTRes *& res, const DTArg * arg , void* func, DCTX(ctx)) = delete;
```

Comment:
We could have a more specialized function pointer here i.e. `(DTRes::VT)(*func)(DTArg::VT)`. 
Problem is that this is currently not supported by *kernels.json*.

Reply:
That's a good point. I copied your comment into the source code (still locally with me), where it is more likely to be found again. From my point of view, it's not urgent, but maybe we will address it at some point.